src/index.js

```diff
@@ -196,6 +199,13 @@ export default class Trigger extends React.Component {
         this.contextMenuOutsideHandler2 = addEventListener(window,
           'blur', this.onContextMenuClose);
       }
+      // close popup on escape keyboard event
+      if (!this.keyDownHandler && props.keyboard) {
+        currentDocument = currentDocument || props.getDocument();
+        this.keyDownHandler = addEventListener(currentDocument,
```

Comment:
In nest Trigger popup, this makes all Trigger close when press ESC.

Reply:
You're right, and I wish it wasn't so.

But this has no simple implementation:
1. Must put the event listener on a single object (e.g. `document`) since it won't fire on a non-focusable element (e.g. `<div>`), thus event bubble/capture is irrelevant.
2. Because of #&#8203;1, the order of event emission is LIFO and for this to work, we need FIFO. This is unchangeable.
3. Even if we do solve this for nested Triggers, because of #&#8203;2, a wrapping `<Modal>` would still react to the keydown event and there's nothing we can do about that (unless we use some custom event manager for all Ant components...).
The reason it works with the `<Select>` is that it uses an `<input>` which doesn't have the restriction noted in #&#8203;1.

The main purpose of this feature is to fix https://github.com/ant-design/ant-design/issues/14699 - the Triggers shouldn't stay open when Modal closes.

Let me know if what you think.

Reply:
Add global event for key board is not a good way cause `ESC` can be triggered on any component. We can't tell which `ESC` should trigger the hide function.
My suggestion is that add key event on child node. If child node can be focusable, it can close by `ESC`. Else do nothing about it.

Reply:
> We can't tell which ESC should trigger the hide function

I'm not sure I understand the risk  - the event listener is only active if the Trigger is open. If any other component in the page is interacted with, the Trigger would close and the listener would be removed.

> If child node can be focusable, it can close by ESC

Yeah that could work.

> Else do nothing about it.

Not worried that it won't be consistent?

Reply:
>> Else do nothing about it.

> Not worried that it won't be consistent?

If we go this route, I think we should make this feature enabled by default and non-configurable. That way there's no expectation, no communication and no risk involved. Just a bug fix.

Reply:
Here is the sample: https://codesandbox.io/s/l5p88216l

ESC to close is good help function, but we need to consider some user may not want close by press ESC everywhere. So I hope just handle can be focused one. (Like button or something else)